### DISCUSSION TITLE
I'm confused about <script setup> in Nuxt 3

### DISCUSSION THREAD

**Discussion by kfconeone** [nuxt/nuxt#17467] [Questions] [answered]
I've been using Vue3 and Nextjs for a while, and I know there is no getServerSidesProps() which can run the code only on server-side before browser fetch the page.

Then I saw the fireship's Nuxt3 intro video, which said that the code in <script setup> is server-side code; that seems a little weird to me, so I made some tests and found that the code in <script setup> did run on server-side but also on client-side, if I run some data-fetching like using axios, it would fetching on both side. I know if I use useAsyncData() or useFetch(), the data-fetching will only occur on server-side, but that just make things more foggy...

```js
await useFetch("https://mysite.com/api/test"); //run only once on server-side
await axios.get("https://mysite.com/api/test"); //run both on server-side & client-side
```

So IMO if I want to run code "only" on server-side, I should use api or middleware, then what's the meaning about code in <script setup> running on server-side?

**Comment by TheAlexLichter** [comment, marked as answer]
Hey 👋  

You can check `process.server` to find out if the setup function is running on server- or client-side if that fits better than, e.g., a server-side Nuxt plugin

**Comment by alexchudnovsky** [comment]
Is there any documentation on this ? I am a bit confused on this as well ?

**Reply by TheAlexLichter** [reply]
There is on several places, e.g. [the Nuxt lifecycle](https://nuxtjs.org/docs/concepts/nuxt-lifecycle/#nuxt-lifecycle) (Nuxt2 related though) or roughly on [the rendering mode page](https://v3.nuxtjs.org/guide/concepts/rendering#universal-rendering)
